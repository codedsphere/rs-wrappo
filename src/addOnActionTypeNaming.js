import { ReducerDefaultActionTypes } from './constants/reducer';

export default function addOnActionTypeNaming(
  ActionTypes,
  isReducer = false
) {
  for (const [actionTypesKey, actionTypesValue] of Object.entries(
    ActionTypes,
  )) {
    if (isReducer) {
      for (const [
        defaultActionTypesKey,
        defaultActionTypesValue,
      ] of Object.entries(ReducerDefaultActionTypes)) {
        ActionTypes[actionTypesKey][
          defaultActionTypesKey
        ] = defaultActionTypesValue;
      }
      ActionTypes[actionTypesKey].namespace = actionTypesKey;
    }

    for (const [innerActionTypesKey, innerActionTypesValue] of Object.entries(
      actionTypesValue,
    )) {
      if (innerActionTypesKey !== 'namespace') {
        ActionTypes[actionTypesKey][
          innerActionTypesKey
        ] = `${actionTypesKey}/${innerActionTypesValue}`;
      }
    }
  }
  if (isReducer) {
    if (window.ActionType) {
      window.ActionType = { ...ActionTypes, ...window.ActionType };
    }
    else {
      window.ActionType = ActionTypes;
    }
    return window.ActionType;
  }
  return ActionTypes;
}
