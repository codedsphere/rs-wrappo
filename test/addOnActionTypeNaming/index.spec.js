import addOnActionTypeNaming from 'addOnActionTypeNaming';
import rootReducer from '../__mocks__/rootReducer';
import rootSaga from '../__mocks__/rootSaga';

describe('addOnActionTypeNaming', () => {
  it('should have a addOnActionTypeNaming', () => {
    const ReducerActionTypes = addOnActionTypeNaming(
      rootReducer,
      true,
    );
    const SagaActionTypes = addOnActionTypeNaming(
      rootSaga,
      false,
    );

    expect(ReducerActionTypes).toMatchSnapshot();
    expect(SagaActionTypes).toMatchSnapshot();
  });
});
