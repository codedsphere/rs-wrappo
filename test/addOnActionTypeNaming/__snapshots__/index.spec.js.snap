// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addOnActionTypeNaming should have a addOnActionTypeNaming 1`] = `
Object {
  "appModel": Object {
    "INITIALIZE_ALL": "appModel/INITIALIZE_ALL",
    "INITIALIZE_STATE": "appModel/INITIALIZE_STATE",
    "REMOVE_STATE": "appModel/REMOVE_STATE",
    "UPDATE_STATE": "appModel/UPDATE_STATE",
    "namespace": "appModel",
  },
  "exampleModel": Object {
    "INITIALIZE_ALL": "exampleModel/INITIALIZE_ALL",
    "INITIALIZE_STATE": "exampleModel/INITIALIZE_STATE",
    "Other_Reducer": "exampleModel/Other_Reducer",
    "REMOVE_STATE": "exampleModel/REMOVE_STATE",
    "UPDATE_STATE": "exampleModel/UPDATE_STATE",
    "namespace": "exampleModel",
  },
  "formModel": Object {
    "INITIALIZE_ALL": "formModel/INITIALIZE_ALL",
    "INITIALIZE_STATE": "formModel/INITIALIZE_STATE",
    "REMOVE_STATE": "formModel/REMOVE_STATE",
    "UPDATE_STATE": "formModel/UPDATE_STATE",
    "namespace": "formModel",
  },
  "themeModel": Object {
    "INITIALIZE_ALL": "themeModel/INITIALIZE_ALL",
    "INITIALIZE_STATE": "themeModel/INITIALIZE_STATE",
    "Other_Reducer": "themeModel/Other_Reducer",
    "REMOVE_STATE": "themeModel/REMOVE_STATE",
    "UPDATE_STATE": "themeModel/UPDATE_STATE",
    "namespace": "themeModel",
  },
  "userModel": Object {
    "INITIALIZE_ALL": "userModel/INITIALIZE_ALL",
    "INITIALIZE_STATE": "userModel/INITIALIZE_STATE",
    "Other_Reducer": "userModel/Other_Reducer",
    "REMOVE_STATE": "userModel/REMOVE_STATE",
    "UPDATE_STATE": "userModel/UPDATE_STATE",
    "namespace": "userModel",
  },
}
`;

exports[`addOnActionTypeNaming should have a addOnActionTypeNaming 2`] = `[Function]`;
